### ISSUE TITLE
Rich Text Editor blocks are not rendering outside of the editor

### ISSUE THREAD

**Issue by mwillebrands** [umbraco/Umbraco-CMS#15675]
### Which Umbraco version are you using? (Please write the *exact* version, example: 10.1.0)

13.1.0

### Bug summary

When adding a block trough the Rich Text Editor, the block is not rendered as content but the following is rendered:
```
<umb-rte-block class="ng-scope ng-isolate-scope" data-content-udi="umb://element/305b2fdb17cb41fca96b43f14f604781"><!--Umbraco-Block--></umb-rte-block>
```

### Steps to reproduce

1. Create an Element Type with a single text field
2. Create an Document Type with an RTE field with the default profile, with a template
3. Change the RTE default profile so you can add blocks, and add the Element Type that you've created in step 1 as allowed block
4. Create a new instance of the document type created in step 2, and insert a block in the RTE field
5. Render the RTE field in your page template and open the page, notice that the block is not rendered and the umb-rte-block element is in the html.

### Expected result / actual result

I expect the block to be rendered with the cshtml file located in Views/Partials/richtext/Components/{elementtype-alias}.cshtml

I've did some research, and the RTE content is as follows:
```
<umb-rte-block class="ng-scope ng-isolate-scope" data-content-udi="umb://element/305b2fdb17cb41fca96b43f14f604781"><!--Umbraco-Block--></umb-rte-block>
```
As you can see in the above sample ```class="ng-scope ng-isolate-scope"``` is added to the umb-rte-block element.
The Regex that is being used to convert the umb-rte-block elements to the views is located in ```Umbraco.Cms.Core.PropertyEditors.ValueConverters.RichTextParsingRegexes.BlockRegex``` and is as follows:
```<umb-rte-block(?:-inline)? data-content-udi=\"(?<udi>.[^\"]*)\"><!--Umbraco-Block--><\\/umb-rte-block(?:-inline)?>```

This regex does not match when the ```class="ng-scope ng-isolate-scope"``` is added, only if the element attributes start with data-content-udi.

I see multiple solutions for this:

- We can alter the regex so it accepts the class attribute 
- The blockpicker in the Rich Text Editor should be adjusted so that it never saves the ```class="ng-scope ng-isolate-scope"``` attribute
- We strip it on save

**Comment by Zeegaan** [comment]
Hello there! Unfortunately, I cannot reproduce this, my block renders out the view.
Can you reproduce this on a clean install? 🤔 
![image](https://github.com/umbraco/Umbraco-CMS/assets/70372949/7afa9073-c0ed-4db1-bc10-de0457032b72)
![image](https://github.com/umbraco/Umbraco-CMS/assets/70372949/cafe3b37-44bc-413b-a521-52a4bba3fdec)

**Comment by mwillebrands** [comment]
Hi,

My apologies, you are correct. This issue only occurs when you have a Rich Text field inside a BlockGrid, and will probably not be related to the causes I've mentioned above. I've attached a sample project with everything synced with USync.
[UmbracoProject1.zip](https://github.com/umbraco/Umbraco-CMS/files/14186389/UmbracoProject1.zip)


You should be able to run this without problems.

Regards,

**Comment by Zeegaan** [comment]
I still cannot reproduce this sadly 😢 

Here's my content:
![image](https://github.com/umbraco/Umbraco-CMS/assets/70372949/411a7609-1515-474c-a1c1-8894da1e6f99)

It renders out correctly:
![image](https://github.com/umbraco/Umbraco-CMS/assets/70372949/6f5f5b5c-e32e-49be-a584-a84264181ad7)


My template looks like this: 
![image](https://github.com/umbraco/Umbraco-CMS/assets/70372949/2e9d20eb-52f5-4792-b3a3-6d3324296db4)

**Comment by mwillebrands** [comment]
@Zeegaan have you seen the attached sample  project? That should have everything to set up a repo case, if not I'll do my best to describe it textually.

**Comment by Zeegaan** [comment]
No totally overlooked that sorry, that had everything i needed! 
Wierd that as you mention, it displays no value, but if you do a `.GetValue` it spits out a value:

![image](https://github.com/umbraco/Umbraco-CMS/assets/70372949/7b58415b-312e-4fad-af6d-5645934de15d)

**Comment by mwillebrands** [comment]
Great to hear, thanks for being able to verify it so quickly. It seems it's not replacing the macro when rendering it from within a BlockGrid. If you have any pointers I'd be happy to have a look if i can fix the issue.

**Comment by Zeegaan** [comment]
Even more wierdness incoming, when doing it on my own site, that works perfectly fine 🙈 I could scream 🤣

**Comment by Zeegaan** [comment]
I did however notice I used SQlite for this, and you were using SqlServer, looking into if that makes the difference 🕵️

**Comment by Zeegaan** [comment]
No dice there either, have tested both databases from scratch now sadly 😢 

Might need to have you try some steps from a clean install to see if you can reproduce this, and maybe a GIF so I'm not doing something you aren't 🙈

**Comment by mwillebrands** [comment]
I'll set it up in a clean project again with SQlite and I'll desribe all the steps taken to get to the endresult, probably this afternoon.

**Comment by mwillebrands** [comment]
Ok so here we go, I'm looking forward to see if you can reproduce it this way, and what differs between our way of setting this up.

# Prerequisites
1. Create a new umbraco project (dotnet new umbraco --name Issue15675 --version 13.1.1)
2. Install Views.zip ([Views.zip](https://github.com/umbraco/Umbraco-CMS/files/14209527/Views.zip))

## 1. Create Element Type
*Name*: CTA Rich Text
*Alias*: ctaRichText
### 1.1 Add property
*Group name*: Details
*Name*: MyProperty
*Alias*: myProperty
*Type*: TextString

## 2. Create Element Type
*Name*: My Block Grid Item
*Alias*: myBlockGridItem
### 2.1 Add property
*Group name*: Details
*Name*: MyRteField
*Alias*: myRteField
*Type*: Richtext editor using default RTE 

## 3. Create Data Type
*Name*: My Block Grid
*Property Editor*: Block Grid
*Blocks*: My Block Grid Item

## 4. Change Data Type for Richtext editor
*Toolbar*: Allow Block
*Available Blocks*: CTA Rich Text

## 5. Create Document Type with Template
*Name*: Test Page
*Alias*: testPage
### 5.1 Add property
*Group name*: Details
*Name*: MyBlockGrid
*Alias*: myBlockGrid
*Type*: Block Grid using configuration "My Block Grid" from step 3
### 5.2 Configure permissions
*Allow as root*: true

## 6. Create content
*Type*: Test page
*Name*: Test
### 6.1 Add item to block grid
*Type*: My Block Grid Item
### 6.2 Configure My block Grid Item
### 6.2.1 Configure field *MyRteField*
1. Insert Block
2. Select CTA Rich Text
3. Set MyProperty to Test

## 7 Save and publish content
## 8 Validate results
Note that the page shows <umb-rte-block instead of the replaced macro.

**Comment by andr317c** [comment]
Hey! I just took a look at this issue. And I was able to reproduce it on version 13.1.1. I also tested it on the unreleased version 13.2.0 and the issue seems to be fixed, the RC for 13.2.0 will be released on Thursday 22/02/2024. You can test this on the source code for version 13. 

Since the issue is fixed on version 13.2.0 I will go ahead and close this. Please reopen this issue if you still encounter this issue on version 13.2.0 😄
